test/realtime/message.test.js

```diff
@@ -1183,5 +1183,119 @@ define(['ably', 'shared_helper', 'async', 'chai'], function (Ably, helper, async
           });
       });
     }
+
+    it('subscribes to filtered channel', function (done) {
+      var testData = [
+        {
+          name: 'filtered',
+          data: 'This message can be filtered',
+          extras: {
+            headers: {
+              name: 'Lorem Choo',
+              number: 26095,
+              bool: true,
+            },
+          },
+        },
+        {
+          name: 'filtered',
+          data: 'Random data with no extras for filter',
+        },
+        {
+          name: 'filtered',
+          data: 'Another filtered message',
+          extras: {
+            headers: {
+              name: 'John Bull',
+              number: 26095,
+              bool: false,
+            },
+          },
+        },
+        {
+          name: 'filtered',
+          data: 'No header on message',
+        },
+        {
+          name: 'end',
+          data: 'last message check',
+        },
+      ];
+
+      var filterOption = {
+        filter: 'name == `"filtered"` && headers.number == `26095`',
+      };
+
+      try {
+        /* set up realtime */
+        var realtime = helper.AblyRealtime({ key: helper.getTestApp().keys[5].keyStr });
+        var rest = helper.AblyRest();
+
+        realtime.connection.on('connected', function () {
+          var rtFilteredChannel = realtime.channels.getDerived('chan', filterOption);
+          var rtUnFilteredChannel = realtime.channels.get('chan');
+
+          var filteredMessages = [];
+          var unFilteredMessages = [];
+          /* subscribe to event */
+          rtFilteredChannel.subscribe('filtered', function (msg) {
```

Comment:
We don't need the client-side name event name filters, let's just look at everything that comes back on the channel, since

```
name == `"filtered"`
```

in the expression will do this server-side.